tiiuae/ghaf
# Refactor net-vm

## Description of changes



Refactoring of netvmExtraModules:
  - remove netvm extra modules file and use pci passthrough module
  - create wifi config service
  - refactor host names to module to allow query ips
  - create reference service module + passthrough to VMs
  - moved dendrite as first reference service

## Checklist for things done






## Testing



### Testing done 
- Build and run on x1
- checked `smcroute` service is running in net-vm 
- build with reference service disabled


### Testing required
- verify build on all platforms
- build and test on x1
- verify dendrite-pinecone + element works
- verify build works without dendrite-pinecone
- verify build works without reference services
